databricks_cli/__init__.py

```diff
@@ -24,6 +24,17 @@
 # pylint:disable=import-error
 # pylint:disable=bare-except
 
+import warnings
+
+
+def issue_deprecation_warning():
+    warnings.warn("the databricks-cli module is deprecated in favor of databricks-sdk-py. Python "
+                  "3.12 will be the last version of Python supported by databricks-cli. Please "
+                  "migrate to databricks-sdk-py as documented in the migration guide: "
+                  "https://docs.databricks.com/en/dev-tools/cli/migrate.html",
+                  DeprecationWarning, stacklevel=3)
```

Comment:
What does the stacklevel mean?

Reply:
Ideally, if folks use the Databricks CLI itself, they don't see these.

Reply:
stacklevel changes the output of warnings.warn. Normally, the warning message starts with the line that the warning.warn call is on, but increasing stacklevel allows the warning to point to the place where the caller uses the library/function in question. In this case, the warning will point to the line with `import databricks_cli`:
<img width="1662" alt="Screenshot 2023-10-05 at 10 41 20" src="https://github.com/databricks/databricks-cli/assets/1850319/2471a0cc-a7b5-4dc5-a4f3-11fc8c6518bd">